conda/activate.py

```diff
@@ -805,7 +799,26 @@ def _get_environment_env_vars(self, prefix):
                     )
                     print(f"variable {dup} duplicated", file=sys.stderr)
                 env_vars.update(prefix_state_env_vars)
-
+        
+        # Remove reserved environment variables and warn if they're being set
+        collect_reserved_vars = []
+        for reserved in RESERVED_ENV_VARS:
+            if reserved in env_vars:
+                # Only warn if the variable is actually being set (not unset)
+                if env_vars[reserved] != CONDA_ENV_VARS_UNSET_VAR:
+                    collect_reserved_vars.append(reserved)
+                # Remove from env_vars regardless
+                env_vars.pop(reserved)
```

Comment:
When I tested the reset logic, I found that vars that are unset are getting the value of `CONDA_ENV_VARS_UNSET_VAR` which means `***unset***` so we'll need to filter that out here, so even if users have unset the value, it'll be filtered out where this method is called, but we don't want to warn them here already, and only remove them from the vars. We can't change the return value of the `_get_environment_env_vars` method since the full list is needed for the deactivation.

Reply:
Cool, I think this is a good change! Thanks for pushing that up. 

I also have this [follow up PR](https://github.com/conda/conda/pull/15354) that warns the user if they are setting/unsetting a reserved environment variable. In that PR, conda will fully remove a reserved env var from the state file if it is unset. Opposed to giving that variable the `***unset***` value.

I think these 2 changes still work well together. So, @jezdez I'm happy to approve/merge if you are :100: